tests/lean/withSetOptionIn.lean

```diff
@@ -0,0 +1,32 @@
+import Lean
+
+/-!
+# `withSetOptionIn`
+
+This test checks that `withSetOptionIn` recurses into the command syntax (`stx[2]`) in
+`set_option ... in <cmd>`.
+
+Prior to #3806, `withSetOptionIn` erroneously recursed into the syntax `in` (`stx[1]`).
+-/
+
+open Lean Elab Command
+
+/-- Trace `foo` when `set_option trace.debug true`. -/
+elab "#trace_debug_foo" : command => do trace[debug] "foo"
+
+/-- Elab `cmd` using `withSetOptionIn`. -/
+elab "#test " cmd:command : command => do
```

Comment:
```suggestion
elab "#test " cmd:command : command =>
```
I think that the `do` is superfluous here.

Reply:
Ah right, nice! :) I also moved the body onto the same line for brevity.